.github/workflows/label_and_milesone_checker.yml

```diff
@@ -29,3 +29,48 @@ jobs:
         run: |
           echo "Please add a milestone to this PR"
           exit 1
+
+  check_recent_mliestone:
+    name: Check recent milestone
+    if: (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'ready to merge') && github.event.pull_request.milestone != null)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check milestone for closest due date
+        env:
+          GH_TOKEN: ${{ github.token }}
+          PR_MILESTONE_NAME: ${{ github.event.pull_request.milestone.title }}
+        run: |
+          # Install GitHub CLI if necessary
+          # sudo apt-get install -y gh
+
+          IFS='/' read -r repoOwner repoName <<< "${{ github.repository }}"
+
+
+          # Fetch the closest future milestone
+          CLOSEST_MILESTONE=$(gh api graphql -f query='
+            query($repoName: String!, $repoOwner: String!) {
+              repository(name: $repoName, owner: $repoOwner) {
+                milestones(states: OPEN, orderBy: {field: DUE_DATE, direction: ASC}, first: 100) {
+                  nodes {
+                    title
+                    number
+                    dueOn
+                  }
+                }
+              }
+          }' --jq '.data.repository.milestones.nodes | map(select(.dueOn >= now)) | .[0].number' -f repoName="$repoName" -f repoOwner="$repoOwner")
```

Comment:
I actually don't think that we should `map(select(.dueOn >= now))` because I think it will happen relatively often that we miss a deadline by a couple of days. And that's *precisely* the time when this check becomes super important. 😅 

Of course, the converse problem is that we forget to close a deadline. But, brainwave 🧠 🌊, what if we add a step to the release action that closes the corresponding milestone??? 🤯

Reply:
I'm not very familiar with this API. So I prefer to wait with experiments until 0.5.0 is released and closed to check if removing this will not catch closed milestones.